eldar/entry.py

```diff
@@ -21,6 +22,19 @@ def __init__(self, query):
         else:
             self.rgx = None
 
+    def generate_search_regex(self, query: str) -> str:
+        """
+        checks whether the query starts or ends with punctuation and handles it
+        """
+        search_query = f"{query}"
+        if search_query[0] not in string.punctuation:
+            search_query = f"\b{search_query}"
+
+        if search_query[-1] not in string.punctuation:
+            search_query = f"{search_query}\b"
+
+        return search_query
```

Comment:
Ok so interesting one, it looks like that since we return an f-string here it treats the `\b` as a byte character rather than the literal \b and it means that the `test_query_results` is failing. The fix here is simple though, you just need to change ` search_query = f"\b{search_query}"` -> ` search_query = fr"\b{search_query}"`, and then similarly with the second one.

Let me know when you're doing this on Monday morning and we can hop on a quick call and I can show you how to runs the tests as I think that will be useful for you debugging in the future!